### ISSUE TITLE
[mxfp8] [numerics] `trtllm_fp8_block_scale_routed_moe` accuracy is slightly worse than `trtllm_fp8_block_scale_moe`

### ISSUE THREAD

**Issue by zianglih** [flashinfer-ai/flashinfer#2676]
@humansand

This SGLang PR integrates FlashInfer v0.6.4 mxfp8 kernels: https://github.com/sgl-project/sglang/pull/19537
However, on single B200 PTQ Qwen 30B A3B MXFP8 gsm8k, `trtllm_fp8_block_scale_routed_moe` accuracy is slightly worse than`trtllm_fp8_block_scale_moe`.

`trtllm_fp8_block_scale_routed_moe` is useful for RL (rollout-routing-replay compatible), which is very numerics sensitive.
```
cd /sgl-workspace/sglang
python3 benchmark/gsm8k/bench_sglang.py --num-shots 8 --num-questions 1209 --parallel 1209 --platinum
```

```
python -m sglang.launch_server --fp8-gemm-backend flashinfer_trtllm --moe-runner-backend flashinfer_trtllm --model zianglih/Qwen3-30B-A3B-Instruct-2507-MXFP8 &
Accuracy: 0.965
Invalid: 0.000
Latency: 8.733 s
Output throughput: 19333.410 token/s
Accuracy: 0.965
Invalid: 0.000
Latency: 8.224 s
Output throughput: 20529.920 token/s
Accuracy: 0.965
Invalid: 0.000
Latency: 8.077 s
Output throughput: 20905.019 token/s
python -m sglang.launch_server --fp8-gemm-backend flashinfer_trtllm --moe-runner-backend flashinfer_trtllm_routed --model zianglih/Qwen3-30B-A3B-Instruct-2507-MXFP8 &
Accuracy: 0.961
Invalid: 0.000
Latency: 10.147 s
Output throughput: 16812.828 token/s
Accuracy: 0.961
Invalid: 0.000
Latency: 10.116 s
Output throughput: 16864.250 token/s
Accuracy: 0.961
Invalid: 0.000
Latency: 9.672 s
Output throughput: 17637.603 token/s
```
Disabling CUDA graph and piecewise CUDA graph does not remove the gap:
```
python -m sglang.launch_server --disable-cuda-graph --disable-piecewise-cuda-graph --fp8-gemm-backend flashinfer_trtllm --moe-runner-backend flashinfer_trtllm_routed --model zianglih/Qwen3-30B-A3B-Instruct-2507-MXFP8 &
Accuracy: 0.961
Invalid: 0.000
Latency: 21.103 s
Output throughput: 8083.506 token/s
Accuracy: 0.961
Invalid: 0.000
Latency: 20.921 s
Output throughput: 8153.735 token/s
Accuracy: 0.961
Invalid: 0.000
Latency: 20.143 s
Output throughput: 8468.786 token/s
```

**Comment by aleozlx** [comment]
do i read correctly these accuracy are at the eval data set level --num-questions 1209

but does not imply the same experts get routed in both cases and we are comparing the moe output?

routed_moe will still run the routing runner in order to prepare some data needed by the finalize kernel, but i guess here even the experts may defer as we use external routers vs internal routers?

**Comment by zianglih** [comment]
Yes this is for end-to-end testing and routed experts are not guaranteed to be the same. I can run some standalone numrics tests later.

**Comment by zianglih** [comment]
I expanded and ran a few flashinfer numerics test and both apis are bitwise identical. I have further confirmed the regression is due to sglang topk is after routing logits topk.
